glue_exp/tools/floodfill_selection/floodfill_selection.py

```diff
@@ -0,0 +1,187 @@
+        pass
+
+    def close(self):
+        pass
+
+
+class FloodfillMode(MouseMode):
+    """
+    Creates ROIs by using the mouse to 'pick' floodfills out of the data
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        super(FloodfillMode, self).__init__(*args, **kwargs)
+
+        self.icon = QtGui.QIcon(os.path.join(ROOT, "glue_floodfill.png"))
+        self.mode_id = 'Floodfill'
+        self.action_text = 'Floodfill'
+        self.tool_tip = 'Define a region of intrest via floodfills'
+        self.shortcut = 'N'
+        print('icon load!')
+
+    # def move(self, event):
+
+    def roi(self, data, slc, data_object):
+        """Caculate an ROI as the floodfill which passes through the mouse
+
+        Parameters
+        ----------
+        data : `numpy.ndarray`
+            The dataset to use
+
+        Returns
+        -------
+        :class:`~glue.core.roi.PolygonalROI` or `None`
+            A new ROI made by the (single) floodfill that passes through the
+            mouse location (and `None` if this could not be calculated)
+        """
+        # here the xy refers to the x and y related to the order of left panel
+
+        x, y = self._event_xdata, self._event_ydata
+        slc = slc
+        return floodfill_to_roi(x, y, data, slc, data_object)
+
+
+def floodfill_to_roi(x, y, data, slc, data_object):
+    """
+    Return a PolygonalROI for the floodfill that passes through (x,y) in data
+
+    Parameters
+    ----------
+    x, y : float
+        x and y coordinate of the point
+    data : `numpy.ndarray`
+        The data
+
+    Returns
+    -------
+    :class:`~glue.core.roi.PolygonalROI`
+        A new ROI made by the (single) floodfill that passes through the
+        mouse location (and `None` if this could not be calculated)
+    """
+
+    if x is None or y is None:
+        return None
+# TODO: replace
+#     xy = point_floodfill(x, y, data)
+    # calculate the threshold and call draw visual
+    # width = event.pos[0] - self.selection_origin[0]
+    # height = event.pos[1] - self.selection_origin[1]
+    # drag_distance = math.sqrt(width**2+height**2)
+    # canvas_diag = math.sqrt(self._vispy_widget.canvas.size[0]**2
+    #                         + self._vispy_widget.canvas.size[1]**2)
+
+    # mask = self.draw_floodfill_visual(drag_distance / canvas_diag)
+    formate_data = np.asarray(data, np.float64)
+    z = slc
+    threshold = 8.
+    print('x, y, z', x, y, z)
+    # coordinate should be integers as index for array
+    mask = floodfill_scipy(formate_data, (z, int(round(y)), int(round(x))), threshold).astype(float) * 5
+
+    if mask is not None:
+        new_mask = np.reshape(mask, data.shape)
+        new_mask = np.ravel(np.transpose(new_mask))
+        print('new_mask', np.sum(new_mask))
+        mark_selected(new_mask, data, data_object)
+    # if xy is None:
+    #     return None
+
+    # p = roi.PolygonalROI(vx=xy[:, 0], vy=xy[:, 1])
+    # return p
```

Comment:
I am not sure if I only update the subset_mode but not create a subset and apply it to the widget, as the `ROI` does, is it possible to get it shown? When does a subset get created?